lalrpop/src/lexer/nfa/mod.rs

```diff
@@ -263,47 +262,42 @@ impl NFA {
             },
 
             HirKind::Repetition(ref r) => {
-                if !r.greedy {
```

Comment:
I took a stab at this because I saw @nikomatsakis comment https://github.com/lalrpop/lalrpop/issues/209#issuecomment-291263299 in the linked issue. 

In my case, I was doing `r"'.*?[^']'"` to capture everything in between single quotes up to the next single quote that does not have a single quote before it and `r"''"` to then grab an empty string without having to or operator and specify non capturing groups in the first regex.

In my case though,that didn't give me what I needed  to parse what I needed, but non-greedy regexes give a decent bit of extra control.